docs/variants/dell_optiplex/recovery.md

```diff
@@ -1,85 +1,210 @@
+</center>
+
+Clip on the `SPI_1` chip. Match pin 1 (`CS`) of the Pomona clip with the first
+pin of `SPI_1` chip, marked with a small dot engraved on the chip.
+
+![Clip connected](../../images/clip_connected.jpg)
+
+## Step 4: Connect RTE
+
+<center>
+![](../../images/rte-boot.jpg)
+</center>
+
+1. Connect J2 Orange Pi Zero system debug output
+2. Power the board and confirm it boots
+3. **Please note** typical convention of USB-UART converter colors is as follows
+    - black - GND
+    - red - +5V
+    - green - TX
+    - white - RX
+4. Connect terminal to RTE and read OS version:
+
+    ```shell
+    sudo minicom -b 115200 -D /dev/ttyUSB0 -o -C /tmp/minicom.cap
+    ```
+
+    - `-b 115200` sets baudrate
+    - `-D /dev/ttyUSB0` points to USB-UART converter device, it can be
+      different if you already have some devices connected or you use different
+      operating system
+    - `-o` skip initialization
+    - `-C /tmp/minicom.cap` capture serial terminal output, if you will have
+      problems with exercises please post this file
+5. Login using following credentials:
+
+    ```shell
+    login: root
+    password: meta-rte
+    ```
+
+## Step 5: Prepare recovery binary
+
+* Get v0.2-rc3 binary
 
 ```bash
-sudo flashrom -p internal -i me -w <optiplex_firmware>
+wget https://cloud.3mdeb.com/index.php/s/8WNEHEFcBGFRK23/download -O dasharo_dell_optiplex_9010_v0.2-rc3.rom
 ```
 
-for example:
+* It has 12MB, so it have to be split
 
 ```bash
-sudo flashrom -p internal -i me -w /tmp/backup.rom
+split -b4M dasharo_dell_optiplex_9010_v0.2-rc3.rom
 ```
 
-output:
+## Step 6: Flash 4MB (BIOS) part
 
 ```bash
-flashrom v1.2-551-gf47ff31 on Linux 5.10.0-9-amd64 (x86_64)
-flashrom is free software, get the source code at https://flashrom.org
-
-Using clock_gettime for delay loops (clk_id: 1, resolution: 1ns).
-Found chipset "Intel Q77".
-Enabling flash write... SPI Configuration is locked down.
-The Flash Descriptor Override Strap-Pin is set. Restrictions implied by
-the Master Section of the flash descriptor are NOT in effect. Please note
-that Protected Range (PR) restrictions still apply.
-Enabling hardware sequencing due to multiple flash chips detected.
-OK.
-Found Programmer flash chip "Opaque flash chip" (12288 kB, Programmer-specific) mapped at physical address 0x0000000000000000.
-Reading ich descriptor... done.
-Using regions: "me", "bios".
-Reading old flash chip contents... done.
-Erasing and writing flash chip... Erase/write done.
-Verifying flash... VERIFIED.
+echo 1 > /sys/class/gpio/gpio405/value
 ```
 
-If you get a warning:
+```bash
+echo 1 > /sys/class/gpio/gpio406/value
+```
 
 ```bash
-WARNING! You may be running flashrom on an unsupported laptop.
+echo 1 > /sys/class/gpio/gpio404/value
 ```
 
-And programmer initialization failed, run command:
+`xac` is third file resulting from previous binary split, so it contains BIOS
+region which in case of Dell OptiPlex 9010/7010 should be flashed to 4MB SPI.
```

Comment:
```suggestion
`xac` is third file resulting from the previous binary split, so it contains Dasharo code and data
which fits into 4MB and in case of Dell OptiPlex 9010/7010 should be flashed to 4MB SPI.
```